master/buildbot/www/hooks/github.py

```diff
@@ -242,6 +245,30 @@ def _get_commit_msg(self, repo, sha):
         msg = data.get('commit', {'message': 'No message field'})['message']
         return msg
 
+    @defer.inlineCallbacks
+    def _get_pr_files(self, repo, number):
+        """
+        Get Files that belong to the Pull Request
+        :param repo: the repo full name, ``{owner}/{project}``.
+            e.g. ``buildbot/buildbot``
+        :param number: the pull request number.
+        """
+        headers = {"User-Agent": "Buildbot"}
+        if self._token:
```

Comment:
I guess if the token is not there, we should return [] directly.

I mean people must provide a token to get pr files.

authentication-less api is heavily rate limited on GH. I don't think it is suitable for CI.

Reply:
This code directly mirrors the function `_get_commit_msg` https://github.com/buildbot/buildbot/blob/master/master/buildbot/www/hooks/github.py#L225-L243

Then that would be a problem as well.

Reply:
This must have been an overlook before. can we fix the two offending places?

Reply:
Github's unauthenticated rate limit is 60/hour. Authenticated it's 5000. [1]
For us at ScummVM, the 60/hour limit is more than enough.

Having the option to not have need a token sped up my development iteration.
This would also need a documentation update, reflection this change [2].

How are we going to resolve this?

[1] https://developer.github.com/v3/#rate-limiting
[2] https://docs.buildbot.net/latest/manual/configuration/wwwhooks.html#github-hook

Reply:
fair enough. We can add a simple warning in the doc.